components/paragraphs/ParagraphResolver.tsx

```diff
@@ -0,0 +1,31 @@
+import React from "react"
+
+import { ParagraphUnion } from "@/lib/graphql/generated/dpl-cms/graphql"
+
+import ParagraphGoVideo from "./ParagraphGoVideo/ParagraphGoVideo"
+import { ParagraphErrorBoundary } from "./paragraphErrorBoundary/paragraphErrorBoundary"
+
+function ParagraphResolver({ paragraphs }: { paragraphs: ParagraphUnion[] }) {
+  const components = {
+    ParagraphGoVideo,
+  }
+
+  return paragraphs.map((paragraph, index) => {
+    const type = paragraph?.__typename
+    if (!type) {
+      return null
+    }
+
+    const DynamicComponentType = components[type as keyof typeof components] || null
+    if (DynamicComponentType === null) return null
+
+    return (
+      <ParagraphErrorBoundary key={index}>
```

Comment:
Nice that you introduce a Paragraph Error boundary.
Is we talked about earlier we should have a clear defined way of treating errors throughout the application.
Please write an ADR about what the idea is behind treating errors in paragraphs. Or if you have a general idea of handling errrors in components with error boundaries the ADR can contain information about that too.